### PR TITLE
feat: Make "Post Preview URLs" action more generic

### PR DESCRIPTION
**Pull request by pekala**
BREAKING CHANGE: The post preview url actions now takes a JSON definition of links to post

Refactors the `post-preview-urls` action to take a JSON array describing the links to post to the PR
Returns the hostname label used for deployment from the cursor deploy action 
Adjust reusable workflows to work with the above updates.

### CODE REVIEW

File: actions/branch-name-to-hostname-label/main.test.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import {branchNameToHostnameLabel} from './main'
+jest.mock('../utils')
+jest.mock('@actions/core')
+
+// reset the counter on mock fn calls after every test
+beforeEach(() => jest.clearAllMocks())
```
Comment (skovhus) [inline actions/branch-name-to-hostname-label/main.test.ts:6]: We should switch to configure this globally: https://jestjs.io/docs/configuration#clearmocks-boolean

File: actions/branch-name-to-hostname-label/main.test.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+describe(`Branch Name to Hostname Label Action`, () => {
+    test(`sanitizes a full git ref into a valid hostname label`, async () => {
```
Comment (skovhus) [inline actions/branch-name-to-hostname-label/main.test.ts:9]: Nitpick: personally I prefer `it` as it reads better than `test`. But very much a stylistic comment. :)

**Review by skovhus — APPROVED** [review]
Looks really good! 👏

File: actions/branch-name-to-hostname-label/main.test.ts (hunk 1 above)
Comment (pekala) [reply actions/branch-name-to-hostname-label/main.test.ts:6]: TIL!

File: actions/cursor-deploy/main.test.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ jest.mock('@actions/github')
 const mockedUtils = jest.mocked(utils)
 
 describe(`Cursor Deploy Action`, () => {
-    test(
+    it(
```
Comment (skovhus) [inline actions/cursor-deploy/main.test.ts:13]: Actually these `test` was better as they describe a test case. 🙃 

So `test` for test case and `it` for sentences that reads well `it sanitizes` vs `test when the action runs`.

Comment (pekala) [reply actions/cursor-deploy/main.test.ts]: word

**Comment by pekala** [comment]
@skovhus I changed the approach - no new action, instead I just return the label from the cursor deploy one - much simpler.

**Comment by pekala** [comment]
@skovhus Since this is mostly the same as you reviewed, just a bit less complicated, I'll go ahead and merge

**Pull request merged**